src/cuda.hpp

```diff
@@ -172,6 +178,7 @@ class GPUWait {
   void signal();
  private:
   int32_t* wait_sync __attribute__((aligned(64)));
+  int32_t* wait_sync_dev_ptr_no_stream_mem_ops __attribute__((aligned(64)));
```

Comment:
Something to consider later would be to see if we can do this without additional memory overhead.

Reply:
With the unified memory stuff, I don't think you actually need to get the device pointer. It's probably the same if I'm reading [this doc](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY.html#group__CUDART__MEMORY_1gc00502b44e5f1bdc0b424487ebb08db0) correctly. This is all also possible with just the driver API, in which case CUdeviceptr is probably what we'd need, but I'm less comfortable with that API.